sql/python/sqlflow/models/example.py

```diff
@@ -78,16 +78,16 @@ def prepare_prediction_column(self, prediction):
         return prediction.argmax(axis=-1)
 
 model = DNNClassifier(feature_columns=feature_columns, hidden_units=[10, 10], n_classes=3)
-model.compile(optimizer=model.default_optimizer(), loss=model.default_loss())
-
 is_training = False
 if is_training:
+    model.compile(optimizer=model.default_optimizer(), loss=model.default_loss())
     model.fit(train_ds, validation_data=val_ds, epochs=model.default_training_epochs(), verbose=0)
+    #model.save_weights('my_model', save_format="h5")
     model.save_weights('my_model.h5')
     print("Done training.")
 else:
     model.predict(test_ds)
-    model.load_weights('my_model.h5')
+    model.load_weights('my_model.h5', by_name=True)
```

Comment:
What is the purpose of `by_name=True`? In the [official document](https://keras.io/models/about-keras-models/):

>  model.load_weights(filepath, by_name=False) loads the weights of the model from a HDF5 file (created by save_weights). By default, the architecture is expected to be unchanged. To load weights into a different architecture (with some layers in common), use by_name=True to load only those layers with the same name.

I am not sure if we are expecting to change the architecture.

Reply:
Removed